tools/site_and_regional/run_neon.py

```diff
@@ -254,20 +195,45 @@ def get_parser(args, description, valid_neon_sites):
             if site not in valid_neon_sites:
                 raise ValueError("Invalid site name {}".format(site))
 
-    if "CIME_OUTPUT_ROOT" in args.case_root:
-        args.case_root = None
-    if args.run_type:
-        run_type = args.run_type
-    else:
-        run_type = "ad"
-    if run_type == "ad":
-        run_length = int(args.ad_length)
-    elif run_type == "postad":
-        run_length = int(args.postad_length)
-    else:
-        run_length = 0
+    if "CIME_OUTPUT_ROOT" in args.output_root:
+        args.output_root = None
+
+    if args.run_length == '0Y':
+        if args.run_type == 'ad':
+            run_length = '200Y'
+        elif args.run_type == 'postad':
+            run_length = '50Y'
+        else:
+            run_length = '4Y'
 
-    return neon_sites, args.case_root, run_type, args.overwrite, run_length, args.base_case_root
+    run_length = parse_isoduration(run_length)
+    if args.base_case_root:
+        base_case_root = os.path.abspath(args.base_case_root)
+
+    return neon_sites, args.output_root, args.run_type, args.overwrite, run_length, base_case_root, args.run_from_postad
+
+def get_isosplit(s, split):
+    if split in s:
+        n, s = s.split(split)
+    else:
+        n = 0
+    return n, s
+
+def parse_isoduration(s):
+    '''
+    simple ISO 8601 duration parser, does not account for leap years and assumes 30 day months
```

Comment:
what is this code being used to do?  I'm assuming it's not used for transient runs?

Reply:
This is used to set the run length.  I don't think that it will be a very commonly used option, but it accept options in the ISO8601 format, so for example 3Y (3 years) or 3Y4M (3 years 4 months), or 7D (7 days).

Reply:
OK, if you run transient case from 2018-2020 will there be 365 days / year (with 366 in leap years)?